### PR TITLE
Commerce 1539 (3/7) Fix potential bugs and permissioning

### PR DESCRIPTION
**Pull request by alecsloan**

### CODE REVIEW

**Comment by alecsloan** [comment]
ci:test

**Comment by alecsloan** [comment]
ci:test

**Comment by alecsloan** [comment]
ci:test

**Comment by alecsloan** [comment]
ci:test

**Comment by alecsloan** [comment]
ci:test

**Comment by alecsloan** [comment]
ci:test

**Comment by alecsloan** [comment]
@marco-leo Rebased

**Comment by alecsloan** [comment]
ci:test

**Comment by alecsloan** [comment]
ci:test

File: commerce-account-service/src/main/java/com/liferay/commerce/account/service/impl/CommerceAccountServiceImpl.java [hunk 1]
```diff
@@ -102,40 +101,40 @@ public CommerceAccount fetchByExternalReferenceCode(
...
-				getUserId(), commerceAccountId);
-
-		if (commerceAccount == null) {
-			throw new PrincipalException.MustHavePermission(
-				getPermissionChecker(), CommerceAccount.class.getName(),
-				commerceAccountId, ActionKeys.VIEW);
-		}
-
-		return commerceAccount;
+		return null;
```
Comment (marco-leo) [inline commerce-account-service/src/main/java/com/liferay/commerce/account/service/impl/CommerceAccountServiceImpl.java]: we are not returning an account now

File: commerce-media-impl/src/main/java/com/liferay/commerce/media/internal/DefaultCommerceMediaResolver.java [hunk 2]
```diff
@@ -143,7 +143,7 @@ public String getUrl(
 		}
 		else {
 			cpAttachmentFileEntry =
-				_cpAttachmentFileEntryLocalService.fetchCPAttachmentFileEntry(
+				_cpAttachmentFileEntryService.fetchCPAttachmentFileEntry(
```
Comment (marco-leo) [inline commerce-media-impl/src/main/java/com/liferay/commerce/media/internal/DefaultCommerceMediaResolver.java]: check secure pattern implemented in ExpandoBridge

Comment (marco-leo) [reply commerce-media-impl/src/main/java/com/liferay/commerce/media/internal/DefaultCommerceMediaResolver.java]: @ethanbustad  in portal secure is used differently ExpandoBridgeImpl.addAttribute

File: commerce-service/src/main/java/com/liferay/commerce/internal/constants/CommerceOrderAvailableOrderConstants.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+ */
+
+package com.liferay.commerce.internal.constants;
+
+import com.liferay.commerce.constants.CommerceOrderConstants;
+
+/**
+ * @author Alec Sloan
+ */
+public class CommerceOrderAvailableOrderConstants {
```
Comment (marco-leo) [inline commerce-service/src/main/java/com/liferay/commerce/internal/constants/CommerceOrderAvailableOrderConstants.java]: @ethanbustad I don't understand why we need to have this class, find bug is suggesting

File: commerce-service/src/main/java/com/liferay/commerce/internal/order/CommerceOrderHttpHelperImpl.java [hunk 4]
```diff
@@ -97,7 +97,9 @@ public CommerceOrder addCommerceOrder(HttpServletRequest httpServletRequest)
 				commerceAccount.getCommerceAccountId(), commerceCurrencyId);
 		}
 
-		setCurrentCommerceOrder(httpServletRequest, commerceOrder);
```
Comment (marco-leo) [inline commerce-service/src/main/java/com/liferay/commerce/internal/order/CommerceOrderHttpHelperImpl.java:100]: we can set current order to null

Comment (marco-leo) [reply commerce-service/src/main/java/com/liferay/commerce/internal/order/CommerceOrderHttpHelperImpl.java:100]: forget

File: commerce-media-impl/src/main/java/com/liferay/commerce/media/internal/DefaultCommerceMediaResolver.java (hunk 2 above)
Comment (alecsloan) [reply commerce-media-impl/src/main/java/com/liferay/commerce/media/internal/DefaultCommerceMediaResolver.java]: It still seems flipped to me. Wouldn't be more clear to use a variable like `checkPermissions` or something like that? Cause at this point if someone sees `secure` as a parameter theres a 50% chance they'll think of it either way.

Comment (alecsloan) [reply commerce-media-impl/src/main/java/com/liferay/commerce/media/internal/DefaultCommerceMediaResolver.java]: for reference, https://www.merriam-webster.com/dictionary/secure 
`1a: free from danger`
which would imply if (secure) then use local service ;)

**Comment by alecsloan** [comment]
@marco-leo fixed and rebased. Retesting.

**Comment by alecsloan** [comment]
ci:test

**Comment by alecsloan** [comment]
ci:test

**Comment by alecsloan** [comment]
ci:stop

**Comment by alecsloan** [comment]
ci:test

**Comment by marco-leo** [comment]
@alecsloan this pull changes too many things. Please just send a small fix containing only the bugs fixing

**Pull request closed without merging**